dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/CompanyUploader.kt

```diff
@@ -126,4 +127,33 @@ class CompanyUploader(
             )
         }
     }
+
+    /**
+     * Updates the ISINs of all companies.
+     * @param leiIsinMap the delta-map with the format "LEI"->"ISIN1,ISIN2,..."
+     */
+    fun updateIsinMapping(leiIsinMap: Map<String, String>) {
+        retryOnCommonApiErrors {
+            for ((lei, isins) in leiIsinMap) {
+                val isinList = isins.split(",").map { it.trim() }
+                val company = companyDataControllerApi.getCompaniesBySearchString(lei)
+                if (company.isNotEmpty()) {
```

Comment:
We should actually check that there is exactly one result. If there are more the first one is not necessarily the one we are looking for.

Reply:
Hold on, I think it would be best to filter `company`/`companies` for something with the required lei and check if that list ist not empty and take the first from these result, because you do not check for the identifier to be the correct LEI, yet anyways.

Reply:
I think LEI is unique. However, the endpoint does not only search for LEI. I'm not sure if there could be a different identifier type with this LEI as value. This would be found by this endpoint as well. I just want to be safe that such an unlikely event can not happen.